controller/pokemon/pokemon.go

```diff
@@ -0,0 +1,49 @@
+package pokemon
+
+import (
+	"net/http"
+	"bootcamp/usecase/pokemon"
+	"bootcamp/service/network"
+	"github.com/gorilla/mux"
+)
+
+/*
+GetPokemon returns a JSON Pokemon array or a Pokemon information
+If URL not contains /{id} returns a Pokemon array
+If URL contains /{id} return the Pokemon for the given index
```

Comment:
If we're dealing with different paths, then each path should have its own controller, you're dealing with two different data flows, so it would be better to separate them into their own functions.

Reply:
*